### ISSUE TITLE
AI: AI features that don't depend on the proprietary Kythera AI

### ISSUE THREAD

**Issue by AreebDurrani** [o3de/o3de#2043]
**Is your feature request related to a problem? Please describe.**
I see that the AI gem featured on the home page is Kythera AI. While it seems like a great tool, unfortunately it is not available under a free and open source license. The license requires one to either pay royalties or buy out the tool for commercial use, and does not allow the source code to be freely shared and modified.

**Describe the solution you'd like**
 The LmbrCentral gem already provides navmesh functionality. I propose we retain this feature and expand on it with behavior trees, spatial queries, and so on. Of course, Kythera will still be available to those who want to use it.

**Describe alternatives you've considered**
At the very least the engine needs navmesh generation and agent tools that don't depend on Kythera. These already seem to be in LmbrCentral so if expanding them isn't possible we should at least resolve to retain these features rather than removing or replacing them

**Additional context**
In short, while it may be compatible with proprietary plugins, a truly open source engine should not depend on them for critical features.

**Comment by slapin** [comment]
I think adding Detour/DetourCrowd libs support from https://github.com/recastnavigation/recastnavigation should not be a problem,
as well as Recast tiled navmesh, these are great navigation/spatial awareness libraries. The behavior tree should be well integrated into engine to be really useful though and high level constructs it operates on should be implemented beforehand...

**Comment by AreebDurrani** [comment]
> I think adding Detour/DetourCrowd libs support from https://github.com/recastnavigation/recastnavigation should not be a problem,
> as well as Recast tiled navmesh, these are great navigation/spatial awareness libraries. The behavior tree should be well integrated into engine to be really useful though and high level constructs it operates on should be implemented beforehand...

Yes I think this is a good idea. We should consider whether we should extend the AI features in LmbrCentral or create a separate open source AI gem.
